File: first-unique-character-in-a-string/first-unique-character-in-a-string.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        res = {}
        
        for ch in s:
            if ch not in res:
                res[ch] = 0
            res[ch] += 1
            
        i = 0
        for ch in s:
            if res[ch] < 2:
                return i
            
            i += 1
            
        return -1
        

File: first-unique-character-in-a-string/first-unique-character-in-a-string.java
class Solution {
    public int firstUniqChar(String s) {
        Set<Character> set = new LinkedHashSet();
        Map<Character, Integer> map = new HashMap();
        
        for( char ch: s.toCharArray()){
            map.put(ch , map.getOrDefault(ch, 0)+1);
        }
        int i =0;
        for( char ch: s.toCharArray()){
            if(map.get(ch) < 2){
                return i;
            }
            i++;
        }
        
        return -1;
    }
}